### PR TITLE
Add email to home

### PR DESCRIPTION
**Pull request by flavius-mester**
https://govukverify.atlassian.net/browse/DCMAW-8650

Screenshot 2024-06-17 at 12 17 14

### CODE REVIEW

File: app/src/main/java/uk/gov/onelogin/ui/home/HomeScreenViewModel.kt [hunk 1]
```diff
@@ -9,14 +9,16 @@ import kotlinx.coroutines.launch
...
 import uk.gov.onelogin.tokens.usecases.SaveToSecureStore
 import uk.gov.onelogin.tokens.usecases.SaveTokenExpiry
 
 @HiltViewModel
 class HomeScreenViewModel @Inject constructor(
     private val tokenRepository: TokenRepository,
     private val saveToSecureStore: SaveToSecureStore,
-    private val saveTokenExpiry: SaveTokenExpiry
+    private val saveTokenExpiry: SaveTokenExpiry,
+    val getEmail: GetEmail
```
Comment (alex-bradbury) [inline app/src/main/java/uk/gov/onelogin/ui/home/HomeScreenViewModel.kt:21]: This should be private surely?

Then we should probably have some `LiveData` to handle the email in the ui

File: app/src/main/java/uk/gov/onelogin/ui/home/HomeScreen.kt [hunk 2]
```diff
@@ -27,11 +27,12 @@ fun HomeScreen(
 ) {
     viewModel.saveTokens(LocalContext.current as FragmentActivity)
     val tokens = viewModel.getTokens()
+    val email = viewModel.getEmail() ?: ""
```
Comment (alex-bradbury) [inline app/src/main/java/uk/gov/onelogin/ui/home/HomeScreen.kt:30]: See below, think this should be `LiveData`

File: app/src/test/java/uk/gov/onelogin/tokens/usecases/GetEmailImplTest.kt [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    fun setUp() {
+        whenever(tokenResponse.idToken).thenReturn(idToken)
+        whenever(tokenRepository.getTokenResponse()).thenReturn(tokenResponse)
+    }
+
+    @Test
+    operator fun invoke() {
+        val emailResponse = sut.invoke()
+        assertEquals(expectedEmail, emailResponse)
+    }
```
Comment (alex-bradbury) [inline app/src/test/java/uk/gov/onelogin/tokens/usecases/GetEmailImplTest.kt:32]: Can we have some tests in here to check null return results if, for example, the id token is null or the id token doesn't contain an email field etc.

Comment (flavius-mester) [reply app/src/test/java/uk/gov/onelogin/tokens/usecases/GetEmailImplTest.kt:39]: good call -> 50d2dcd248ea1121c47ddb635f7ab7d363638768

File: app/src/main/java/uk/gov/onelogin/ui/home/HomeScreenViewModel.kt (hunk 1 above)
Comment (flavius-mester) [reply app/src/main/java/uk/gov/onelogin/ui/home/HomeScreenViewModel.kt]: as we discussed -> 1d72c76af5b623681ee7d7cd31d46f289a480372

File: app/src/main/java/uk/gov/onelogin/ui/home/HomeScreen.kt (hunk 2 above)
Comment (flavius-mester) [reply app/src/main/java/uk/gov/onelogin/ui/home/HomeScreen.kt]: same 1d72c76af5b623681ee7d7cd31d46f289a480372

**Review by alex-bradbury — APPROVED** [review]

**Pull request merged**